### PR TITLE
Use costmap instead of OccupancyGrid from server

### PR DESCRIPTION
**Pull request by cesar-lopez-mar**
First plan made with FCPP in ROS2! :)

FCPP_ROS2_firstplan

### CODE REVIEW

**Comment by AlaaAlassi** [comment]
Very nice work Cesar. 
The package compiles with no problems, I can't test the code because we need to integrate the node into jax launch files.

File: src/full_coverage_path_planner.cpp [hunk 1]
```diff
@@ -191,17 +191,18 @@ namespace full_coverage_path_planner
...
                                           geometry_msgs::msg::PoseStamped const &realStart,
                                           Point_t &scaledStart)
   {
     uint ix, iy, nodeRow, nodeColl;
-    uint32_t nodeSize = dmax(floor(toolRadius / cpp_grid_.info.resolution), 1);       // Size of node in pixels/units
-    uint32_t robotNodeSize = dmax(floor(robotRadius / cpp_grid_.info.resolution), 1); // RobotRadius in pixels/units
-    uint32_t nRows = cpp_grid_.info.height, nCols = cpp_grid_.info.width;
+    uint32_t nodeSize = dmax(floor(toolRadius / cpp_costmap->getResolution()), 1);       // Size of node in pixels/units
+    uint32_t robotNodeSize = dmax(floor(robotRadius / cpp_costmap->getResolution()), 1); // RobotRadius in pixels/units
+    uint32_t nRows = cpp_costmap->getSizeInCellsY(), nCols = cpp_costmap->getSizeInCellsX();
```
Comment (Timple) [inline src/full_coverage_path_planner.cpp:204]: May I now nitpick about all declerations on it's own line? ;-)

File: src/full_coverage_path_planner.cpp [hunk 2]
```diff
@@ -210,15 +211,18 @@ namespace full_coverage_path_planner
...
 
     // Save map origin and scaling
-    tile_size_ = nodeSize * cpp_grid_.info.resolution; // Size of a tile in meters
-    grid_origin_.x = cpp_grid_.info.origin.position.x; // x-origin in meters
-    grid_origin_.y = cpp_grid_.info.origin.position.y; // y-origin in meters
+    tile_size_ = nodeSize * cpp_costmap->getResolution(); // Size of a tile in meters
+    double originX, originY;
+    cpp_costmap->mapToWorld(0, 0, originX, originY);
+    grid_origin_.x = originX;
+    grid_origin_.y = originY;
```
Comment (Timple) [inline src/full_coverage_path_planner.cpp:218]: Why the intermediate step?

```suggestion
    cpp_costmap->mapToWorld(0, 0, grid_origin_.x, grid_origin_.y);
```

**Review by Timple — COMMENTED** [review]
Awesome, a working version already!

File: src/full_coverage_path_planner.cpp (hunk 1 above)
Comment (cesar-lopez-mar) [reply src/full_coverage_path_planner.cpp]: you may ;) I also changed to type size_t in this function

File: src/full_coverage_path_planner.cpp (hunk 2 above)
Comment (cesar-lopez-mar) [reply src/full_coverage_path_planner.cpp]: there was a float / double issue. just change the type of object. There are more places were float is used. In the type of processors we use, should we consider floats at all?

Comment (Timple) [reply src/full_coverage_path_planner.cpp]: I would just go for doubles everywhere. Doesn't have much impact on an x64 machine.

**Review by Timple — APPROVED** [review]

**Pull request merged**